src/libstd/sys_common/backtrace.rs

```diff
@@ -101,50 +101,75 @@ fn filter_frames(frames: &[Frame],
+        "_ZN3std9panicking",
+        "ZN3std9panicking",
+        "std::panicking",
+        "_ZN4core9panicking",
+        "ZN4core9panicking",
+        "core::panicking",
+        "_ZN4core6result13unwrap_failed",
+        "ZN4core6result13unwrap_failed",
+        "rust_begin_unwind",
+        "_ZN4drop",
+        "mingw_set_invalid_parameter_handler",
+    ];
+    static MAGICS_END: [&'static str; 17] = [
+        "_ZN3std9panicking",
+        "ZN3std9panicking",
+        "std::panicking",
+        "_ZN4core9panicking",
+        "ZN4core9panicking",
+        "core::panicking",
+        "_ZN3std2rt10lang_start",
+        "ZN3std2rt10lang_start",
+        "__rust_maybe_catch_panic",
+        "_rust_maybe_catch_panic",
+        "__libc_start_main",
+        "__rust_try",
+        "_start",
+        "BaseThreadInitThunk",
+        "__scrt_common_main_seh",
+        "_ZN4drop",
+        "mingw_set_invalid_parameter_handler",
+    ];
+
     let mut skipped_before = 0;
     for (i, frame) in frames.iter().enumerate() {
         skipped_before = i;
-        let mut skip = false;
-
+        let mut break_ = false;
         let _ = resolve_symname(*frame, |symname| {
             if let Some(mangled_symbol_name) = symname {
-                let magics_begin = [
-                    "_ZN3std3sys3imp9backtrace",
-                    "_ZN3std10sys_common9backtrace",
-                    "_ZN3std9panicking",
-                    "_ZN4core9panicking",
-                    "rust_begin_unwind",
-                    "_ZN4core6result13unwrap_failed",
-                ];
-                if !magics_begin.iter().any(|s| mangled_symbol_name.starts_with(s)) {
-                    skip = true;
+                if !MAGICS_BEGIN.iter().any(|s| mangled_symbol_name.starts_with(s)) {
+                    break_ = true;
                 }
             }
             Ok(())
         }, context);
-
-        if skip {
-            break;
+        if break_ {
+            break
         }
     }
 
     let mut skipped_after = 0;
-    for (i, frame) in frames.iter().rev().enumerate() {
+    for (i, frame) in frames[skipped_before..].iter().rev().enumerate() {
+        skipped_after = i;
+        let mut break_ = false;
         let _ = resolve_symname(*frame, |symname| {
             if let Some(mangled_symbol_name) = symname {
-                let magics_end = [
-                    "_ZN3std9panicking3try7do_call",
-                    "__rust_maybe_catch_panic",
-                    "__libc_start_main",
-                    "__rust_try",
-                    "_start",
-                ];
-                if magics_end.iter().any(|s| mangled_symbol_name.starts_with(s)) {
-                    skipped_after = i + 1;
+                if !MAGICS_END.iter().any(|s| mangled_symbol_name.starts_with(s)) {
+                    break_ = true;
                 }
             }
             Ok(())
         }, context);
+        if break_ {
+            break
+        }
```

Comment:
Yeah, that's what I meant. I did the current method because I didn't have a cross-platform list of symbols. Your method seems more robust, at least on Tier-1.